From b1b4aa71631502625ed9de6e70ffdf8cc091d53d Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Tue, 19 Aug 2014 15:27:37 -0400
Subject: [tools] perf/record: Use correct return type for write()

Message-id: <1408462094-14194-289-git-send-email-jolsa@redhat.com>
Patchwork-id: 88296
O-Subject: [PATCH RHEL7.1 BZ1131394 288/325] perf record: Use correct return type for write()
Bugzilla: 1131394
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1131394
https://bugzilla.redhat.com/show_bug.cgi?id=1131394

upstream
========
commit 410f17860318f36b999a90138f50350e3dc49933
Author: Adrian Hunter <adrian.hunter@intel.com>
Date: Fri Nov 8 12:27:50 2013 +0200

description
===========
write() returns a 'ssize_t' not an 'int'.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/builtin-record.c b/tools/perf/builtin-record.c
index fc68b26..8f5af32 100644
--- a/tools/perf/builtin-record.c
+++ b/tools/perf/builtin-record.c
@@ -81,7 +81,7 @@ static int write_output(struct perf_record *rec, void *buf, size_t size)
  struct perf_data_file *file = &rec->file;
 
  while (size) {
-  int ret = write(file->fd, buf, size);
+  ssize_t ret = write(file->fd, buf, size);
 
   if (ret < 0) {
    pr_err("failed to write perf data, error: %m\n");
-- 
1.7.1